app/controllers/concerns/api/v3/sync_to_user.rb

```diff
@@ -61,11 +64,18 @@ def current_facility_processed_since
     end
 
     def force_resync?
+      Rails.logger.info "Resync token modified in resource #{controller_name}" if resync_token_modified?
+      Rails.logger.info "Sync region modified in resource #{controller_name}" if sync_region_modified?
+      resync_token_modified? || sync_region_modified?
+    end
+
+    def resync_token_modified?
       process_token[:resync_token] != resync_token
     end
 
-    def resync_token
-      request.headers["HTTP_X_RESYNC_TOKEN"]
+    def sync_region_modified?
+      return if requested_sync_region_id.blank?
+      process_token[:sync_region_id] != requested_sync_region_id
```

Comment:
Would a datadog metric here be useful to gauge how often nurses switch blocks?

Reply:
Yes! A follow up PR that adds metrics is in the works 😄